src/nvim/generators/gen_options.lua

```diff
@@ -1,12 +1,15 @@
 local options_file = arg[1]
 
+local gen_values = options_file:find('options_values') and true or false
+
```

Comment:
I don't like this pattern of using the same generator for 2 completely different purposes. Use a separate generator file for this, similar to what `gen_options_enum` does

Reply:
These are not "complete different purposes". And why is `gen_options_enum.lua` in a separate file?

Reply:
> These are not "complete different purposes".

The script acts does two very different things depending on which file it's being called on, that seems different to me. A script's behavior depending on the filename is certainly not ideal.

> And why is `gen_options_enum.lua` in a separate file?

Precisely because it does a completely different thing. `gen_options` generates the `options[]` array, `gen_options_enum` generates the option index enum (using information from the `options[]` array, so order matters here) and the hash table for `findoption()`

Reply:
this PR has not been processed for a long time, some code may be outdated. the conflict is resolved but seems like test failed. If you have new ideas, you can continue this PR.